sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreCatalogSuite.scala

```diff
@@ -47,6 +47,17 @@ class HiveMetastoreCatalogSuite extends SparkFunSuite with TestHiveSingleton {
     logInfo(df.queryExecution.toString)
     df.as('a).join(df.as('b), $"a.key" === $"b.key")
   }
+
+  test("SPARK-13454: drop a table with a name starting with underscore") {
+    hiveContext.range(10).write.saveAsTable("_spark13454")
+    hiveContext.sql("drop table `_spark13454`")
+
+    hiveContext.range(10).write.saveAsTable("_spark13454")
+    hiveContext.sql("drop table default.`_spark13454`")
+
+    hiveContext.range(10).write.saveAsTable("spark13454")
+    hiveContext.sql("drop table spark13454")
+  }
```

Comment:
Let's add test case that

1. drops temporary tables
2. drops persistent and temporary tables with the same name